jaeger-jaxrs2/src/test/java/com/uber/jaeger/propagation/SpanInfo.java

```diff
@@ -49,8 +51,12 @@ public SpanInfo(Span span) {
     String stringContext = span.getContext().contextAsString();
     traceID = stringContext.substring(0, stringContext.indexOf(':'));
     baggage = span.getBaggageItem(FilterIntegrationTest.BAGGAGE_KEY);
-    sampled =
-        (stringContext.substring(stringContext.lastIndexOf(':') + 1).getBytes()[0] & (byte) 1) == 1;
+    byte byteAfterColon =
+        stringContext
+                .substring(stringContext.lastIndexOf(":") + 1)
+                .getBytes(StandardCharsets.UTF_8)[
+            0];
```

Comment:
this code doesn't make sense (I know you didn't write it, but let's fix it). The value after the last `:` is a hex-encoded byte, not a raw byte converted to char. So we should be simply parsing it as int 

```
String lastNumber = stringContext.substring(stringContext.lastIndexOf(":") + 1);
int flags = Integer.parseInt(lastNumber);
sampled = flags & 1 == 1;
```

or even simpler, can't we just cast `span.getContext()` to jaeger's SpanContext and use `isSampled()`?